src/python/pants/backend/python/rules/coverage.py

```diff
@@ -263,13 +264,14 @@ async def generate_coverage_reports(
 
     pex_processes = []
     report_types = []
+    result_snapshot = await Get(Snapshot, Digest, merged_coverage_data.coverage_data)
```

Comment:
this probably belongs in the merge_coverage_data() rule and the snapshot should be one of the attributes on `MergedCoverageData` @Eric-Arellano wdyt?

Reply:
Yeah, that seems like a good refactor. `MergedCoverageData` is defined in this file, so that's a simple fix.